File: android/src/main/java/com/margelo/nitro/visioncamerafacedetection/HybridTensorFactory.kt
package com.margelo.nitro.visioncamerafacedetection

import androidx.annotation.Keep
import com.facebook.proguard.annotations.DoNotStrip

@DoNotStrip
@Keep
class HybridTensorFactory : HybridTensorFactorySpec() {

  override fun initTensor(modelName: String): String {
    return HybridTensor().initTensor(modelName)
  }

  override fun detectFromBase64(options: TensorFaceOptions): Variant_HybridFaceSpec_NullType {
    return HybridTensor().detectFromBase64(options)
  }
}

File: ios/HybridTensorFactory.swift
import NitroModules
import VisionCamera

class HybridTensorFactory: HybridTensorFactorySpec {
  func initTensor(modelName: String) throws -> String {
    return try HybridTensor().initTensor(modelName: modelName)
  }
  
  func detectFromBase64(options: TensorFaceOptions) throws -> Variant__any_HybridFaceSpec__NullType {
    return try HybridTensor().detectFromBase64(options: options)
  }
}
